src/GslibIO.jl

```diff
@@ -155,4 +196,58 @@ function save(file::File{format"GSLIB"}, sdata::AbstractData)
        propnames=vars)
 end
 
+# low level function for saving `data` to a legacy GSLIB format using `varnames` as variable names
+function save_legacy(filename::AbstractString, data::AbstractMatrix, varnames::NTuple, header::AbstractString, na)
+  @assert size(data, 2) == length(varnames) "Invalid data for the specified variable names"
+  nvars = size(data, 2)
+
+  open(filename, "w") do f
+    write(f, "$header\n")
+
+    write(f, "$nvars\n")
+    for v in varnames
+      write(f, "$v\n")
+    end
+
+    # handle missing values
+    replace!(data, NaN=>na)
+
+    writedlm(f, data, ' ')
+  end
+end
+
+"""
+    save_legacy(file, sdata)
+
+Save spatial data `sdata` to `filename` using standard GSLIB format. It replaces NaNs with `na` values
+and it uses `coordnames` as the given variable name to coordinates
+"""
+function save_legacy(filename::AbstractString, sdata::AbstractData; 
+                     coordnames=(:x, :y, :z), na=-999.0, header="# This file was generated with GslibIO.jl")
+  table = values(sdata)
+  sdomain = domain(sdata)
+  
+  if sdomain isa PointSet
+    # add `coordinates` to data and `coordnames` to `varnames`
+    coords = coordinates(sdomain)
+    cdim = size(coords, 1)
+    @assert cdim <= length(coordnames) "The length of coordinate names must be equal or greater than the coordinate dimension"
+
+    varnames = cat([String(v) for v in coordnames[1:cdim]], names(table), dims=1)
+    data = hcat(transpose(coords), Array(table))
+  elseif sdomain isa RegularGrid
+    # a regular grid does not need to save coordinates
+    varnames = names(table)
+    data = Matrix(table)
+  else
+    error("Only PointSet and RegularGrid can be saved to the legacy GSLIB format")
+  end  
+
+  save_legacy(filename, data, varnames, header, na)
+end
+
+# This variant converts `varnames` from an Array to NTuple
+save_legacy(filename::AbstractString, data::AbstractMatrix, varnames::AbstractArray, header, na) =
+    save_legacy(filename, data, NTuple{size(varnames, 1)}(Symbol.(varnames)), header, na)
```

Comment:
```suggestion
```
I don't think we need a version with varnames as array.

Reply:
But users may need it. It does not hurt I think.

Reply:
This is inconsistent with the attempt to use tuples for these parameters everywhere in the codebase. If users are using arrays for the varnames, they need to be educated to always use tuples. This saves noise in the codebase supporting multiple versions. I already did a major refactoring of the codebase to remove occurrences of arrays for this.

Reply:
Let's remove it